ksqldb-examples/src/main/java/io/confluent/ksql/datagen/ProducerFactory.java

```diff
@@ -50,8 +54,13 @@ static DataGenProducer getProducer(
     final SerializerFactory<Struct> keySerializerFactory =
         keySerializerFactory(keyFormat, ksqlConfig, srClient);
 
+    final Map<String, String> formatInfoProperties = new HashMap<>();
+    if (valueDelimiter != null) {
+      formatInfoProperties.put(DelimitedFormat.DELIMITER, valueDelimiter.toString());
```

Comment:
might want to check `valueFormat` is delimited before setting this, and throw an error if its not.

Reply:
ack, done